vernac/g_vernac.mlg

```diff
@@ -113,7 +113,8 @@ GRAMMAR EXTEND Gram
     ]
   ;
   attribute:
-    [ [ k = ident ; v = attr_value -> { Names.Id.to_string k, v } ]
+    [ [ k = ident ; v = attr_value -> { Names.Id.to_string k, v }
+      | "using" ; v = attr_value -> { "using", v } ]
```

Comment:
If I understand correctly, this change was needed because `using` is defined as a keyword in the Ltac plugin.
In consequence, this makes `using` a keyword even in `-noinit` mode.
Anyway, I would have preferred a more robust change allowing the use of any keyword as an attribute name in the context of `#[ ... ]`. Wouldn't this have been possible?

Reply:
> a more robust change allowing the use of any keyword as an attribute name in the context of #[ ... ]. Wouldn't this have been possible?

Not without a significant change of the current (and standard) decoupling of the lexing/parser phases.

In the present case, it should work with `IDENT "using"` instead and this wouldn't declare a keyword until some other component does it.

Reply:
Good suggestion. Let's do this fix.

Reply:
Hum, isn't "using" always a keyword? I mean, maybe technically it declared in Ltac1's file, I don't know, but to me `Proof using` is vanilla Coq, not something you can unload with -noinit.

Reply:
This is the usual discussion, but, technically, the parser does not require the `using` of `Proof using` to be a keyword for the lexer. Then, of course, we may want to make it a reserved word independently of parsing reasons.

Reply:
I was not raising "the usual discussion" about what should be keywords.